src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts

```diff
@@ -62,11 +80,15 @@ export abstract class WebSocketCloverTransport extends CloverTransport implement
     }.bind(this);
 
     public reconnect(): void {
-        if (this.shutdown) {
-            this.logger.debug("Not attempting to reconnect, shutdown...");
-            return;
-        }
-        setTimeout(this.reconnector, this.reconnectDelay);
+		// If we are already reconnecting, do not start another.
+		if(!this.reconnecting) {
+			this.setReconnecting(true);
+			if (this.shutdown) {
+				this.logger.debug("Not attempting to reconnect, shutdown...");
+				return;
+			}
+			setTimeout(this.reconnector, this.reconnectDelay);
+		}
```

Comment:
Do we need a debug log to show some of the reasons calling reconnect will not do anything.... if shutdown is set, or if already reconnecting?